internal/serialization/serialization.go

```diff
@@ -108,19 +110,19 @@ func (s *Service) WriteObject(output pubserialization.DataOutput, object interfa
 }
 
 func (s *Service) ReadObject(input pubserialization.DataInput) interface{} {
-	serializerID := input.ReadInt32()
-	serializer := s.registry[serializerID]
+	typeID := input.ReadInt32()
+	serializer := s.lookupBuiltinDeserializer(typeID)
+	if serializer == nil {
+		serializer = s.registry[typeID]
+	}
+	if serializer == nil {
```

Comment:
This is a micro-optimization but would it be ok to move this if check into the if check above so that we don't perform two null checks for the happy scenario, where we find built-in serializer?

Reply:
OK, I've refactored that part of the code at: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/646f1893fc76711d3ffecc2c6d09d0253044d316